rbac/api/models.py

```diff
@@ -94,3 +94,20 @@ def username(self, value: Optional[str]) -> None:
     bearer_token: str = ""
     client_id: str = ""
     is_service_account: bool = False
+
+
+class FilterQuerySet(models.QuerySet):
+    """Queryset for filtering."""
+
+    _public_tenant = None
+
+    @classmethod
+    def _get_public_tenant(cls) -> Tenant:
+        """Get or set public tenant."""
+        if cls._public_tenant is None:
+            cls._public_tenant = Tenant.objects.get(tenant_name="public")
```

Comment:
@astrozzc not sure what your thoughts are how this and #1368 might work together? Even without the cache if we tried to model it consistently?

Reply:
Yea, we can update here once the cache is available by updating here to: Tenant.objects.get_public_tenant()

Reply:
yes, maybe it will be enough to have just in-memory memoization https://github.com/RedHatInsights/insights-rbac/pull/1368#issuecomment-2553786125, I am curious on your insights.

Reply:
what about moving the memoization to `@staticmethod` on the Tenant model? (or in follow-up PR?)

Reply:
oh, right, maybe it will be enough to have just in-memory memoization. good idea, moved to the Tenant model